plugins/legacy-plugin-chart-paired-t-test/src/TTestTable.jsx

```diff
@@ -166,6 +166,11 @@ class TTestTable extends React.Component {
   render() {
     const { data, metric, groups } = this.props;
     const { control, liftValues, pValues } = this.state;
+
+    if (!Array.isArray(groups) || groups.length === 0) {
+      throw Error('Group by param is required');
+    }
```

Comment:
You can probably add `validators: [validateNonEmpty]` to the chart control, too:

```ts
      controlSetRows: [
        ['metrics'],
        ['adhoc_filters'],
        [
          {
            name: 'groupby',
            override: {
              validators: [validateNonEmpty],
            },
          },
        ],
```

Reply:
Good point! Done.
Is there any documentation for control panel config besides `ControlPanelConfig` interface?

Reply:
There are some comments in `chart-controls`: https://github.com/apache-superset/superset-ui/blob/6b9cc586b3d8b56a53033b34e861f3f4a9e33a10/packages/superset-ui-chart-controls/src/types.ts#L149

But it's not very easy to find. We should probably start writing better README's.